src/components/Datepicker/__tests__/Datepicker.spec.js

```diff
@@ -31,303 +36,300 @@ describe('Datepicker', () => {
 
   describe('when open', () => {
 
-    const props = { ...initialProps, open: true };
+    let component;
+
+    beforeEach(() => {
+      vi.useFakeTimers();
+      vi.setSystemTime(modelValue);
+
+      component = renderComponent({ props: { ...initialProps, open: false } });
+      const { rerender } = component;
+
+      // immediately use real timers as soon as the component has been rendered as there are issues
+      // with using fake timers with event handler emulators (i.e. fireEvent and userEvent)
+      // but we need the fake timers just long enough to render the component with the mocked system time
+      // which ensures the minDate and maxDate default values are set correctly
+      vi.useRealTimers();
+
+      rerender({ open: true });
```

Comment:
Instead of using `open: true` in the initial `renderComponent` I'm starting with it `open: false`.  By re-rendering after the Fake Timers have been disabled it allows the `focus()` event to fire which the Fake Timers was interfering with.  This allows the focused date to actually have focus and keyboard events to be executed properly.  This probably also more accurately mimics how we really use the `Datepicker` component.